torch/_inductor/ir.py

```diff
@@ -625,20 +625,24 @@ def outer_reduction_splits(reduction_numel_hint, numel_hint):
             if isinstance(r, sympy.Expr) and not isinstance(r, sympy.Number)
         ]
         index = None
-        for md in read_writes.reads:
+        for md in sorted(read_writes.reads):
             if all([r in md.index.free_symbols for r in range_vars]):
                 index = md.index
                 break
         if not index:
             # TODO determine splits when all inputs are broadcasted
             return ReductionHint.DEFAULT, 1
-        for md in read_writes.writes:
+        for md in sorted(read_writes.writes):
```

Comment:
Not sure if this change would fix the issue: https://github.com/pytorch/torchdynamo/issues/1653

Reply:
Yes it will